src/routes/docs/tutorials/nextjs/step-5/+page.markdoc

```diff
@@ -0,0 +1,278 @@
+| amount      | Integer | Yes      |                |
+| type        | Enum    | Yes      | Paid, Received |
+
+Now that we have created a collection, we have to add permissions to the collection. Go to the settings tab, scoll to the permissions sections add the following permissions:
+
+{% only_dark %}
+![Create project screen](/images/docs/tutorials/dark/collection-permissions.png)
+{% /only_dark %}
+{% only_light %}
+![Create project screen](/images/docs/tutorials/collection-permissions.png)
+{% /only_light %}
+
+You can also add permission for a specific user or a team.
+
+Now you will get a database ID and collection ID. Copy the these two and save it in the `/src/config/appwrite.js` file.
+
+```js
+export const EXPENSE_DATABASE_ID = "<YOUR_DATABASE_ID>";
+export const EXPENSE_COLLECTION_ID = "<YOUR_COLLECTION_ID>";
+```
+
+# Add new expense {% #add-new-expense %}
+Now that we have a collection, we can add a new expense to the database.
+
+First we need to create a new API endpoint to add an expense. Create a new file `/src/app/api/expense/route.js` and add the following code.
+
+```js
+import { ID } from "appwrite";
+import { headers } from "next/headers";
+import { NextResponse } from "next/server";
+
+import {
+  EXPENSE_COLLECTION_ID,
+  EXPENSE_DATABASE_ID,
+  account,
+  database,
+} from "@/config/appwrite";
+
+// Post - create expense
+export async function POST(req) {
+  const headersList = headers();
+  const jwt = headersList.get("jwt");
+
+  if (!jwt) {
+    return new NextResponse("Unauthorized", { status: 401 });
+  }
+  account.client.setJWT(jwt);
+
+  const body = await req.json();
+  const { userId, title, amount, type } = body;
+  if (!userId || !title || !amount || !type) {
+    return new NextResponse("Missing fields", { status: 401 });
+  }
+  try {
+    await database.createDocument(
+      EXPENSE_DATABASE_ID,
+      EXPENSE_COLLECTION_ID,
+      ID.unique(),
+      {
+        userId: userId,
+        title: title,
+        amount: amount,
+        type: type,
+      }
+    );
+    return new NextResponse("Saved successfully", { status: 200 });
+  } catch (error) {
+    console.log(error);
+    return new NextResponse("Internal server error", {
+      status: 500,
+      error: error.message,
+    });
+  }
+}
+```
+
+We have used Appwrite `createDocument` method to add a new document to the database. The method takes the following parameters: `databaseId`, `collectionId`, `documentId`, `data`. 
+
+The `databaseId` is the ID of the database we want to add the document to. The `collectionId` is the ID of the collection we want to add the document to. The `documentId` is the ID of the document we want to add. We have used the `ID.unique()` method to generate a unique ID for the document. The `data` is an object that contains the data we want to add to the document.
+
+We have created a new API endpoint that will add a new expense to the database. The endpoint will be available at `/api/expense`.
```

Comment:
```suggestion
In this endpoint, we use the `createDocument` method to add a new document to the database, which accepts the following parameters: `databaseId`, `collectionId`, `documentId`, `data`. 

In Appwrite, documents live in a collection, which in turn live in a database. So to add a document, we need its ID, as well as the ID of its containing collection, and the ID of the database that contains said collection. We also need the data that the document contains.

Since we're creating a new document, it doesn't yet have an ID. We can use the `ID.unique()` method to generate a unique one for it.

The document created by this endpoint represents a new expense. The endpoint is now available at `http://localhost:3000/api/expense`.
```